src/lib/coda_intf/staged_ledger_intf.ml

```diff
@@ -299,59 +299,65 @@ module type Transaction_snark_scan_state_generalized_intf = sig
 
   val empty : unit -> t
 
-  val capacity : t -> int
-
   val fill_work_and_enqueue_transactions :
        t
     -> Transaction_with_witness.t list
     -> transaction_snark_work list
-    -> (ledger_proof * transaction list) option Or_error.t
+    -> ((ledger_proof * transaction list) option * t) Or_error.t
 
   val latest_ledger_proof :
     t -> (Ledger_proof_with_sok_message.t * transaction list) option
 
   val free_space : t -> int
 
-  val next_k_jobs : t -> k:int -> Available_job.t list Or_error.t
-
-  val next_jobs : t -> Available_job.t list Or_error.t
-
-  val next_jobs_sequence : t -> Available_job.t Sequence.t Or_error.t
-
   val base_jobs_on_latest_tree : t -> Transaction_with_witness.t list
 
-  val is_valid : t -> bool
-
   val hash : t -> staged_ledger_aux_hash
 
-  val staged_transactions : t -> ledger_undo list
-
-  val all_transactions : t -> transaction list Or_error.t
+  (** All the staged transactions in the reverse order of their application (Latest first)*)
+  val staged_transactions_with_info_rev : t -> ledger_undo list
 
-  val extract_from_job :
-       Available_job.t
-    -> ( ledger_undo * transaction_snark_statement * transaction_witness
-       , ledger_proof * ledger_proof )
-       Either.t
-
-  val copy : t -> t
+  (** All the transactions in the order in which they were applied*)
+  val staged_transactions : t -> transaction list Or_error.t
 
+  (** Available space and the corresponding required work-count in one and/or two trees (if the slots to be occupied are in two different trees)*)
   val partition_if_overflowing : t -> Space_partition.t
 
   val statement_of_job : Available_job.t -> transaction_snark_statement option
 
-  val current_job_sequence_number : t -> int
-
   val snark_job_list_json : t -> string
 
-  val all_work_to_do :
-    t -> transaction_snark_statement list Sequence.t Or_error.t
+  (** All the proof bundles *)
+  val all_work_statements : t -> transaction_snark_statement list Sequence.t
```

Comment:
Why do some use sequence and some list? Also maybe we should have a bundle type alias?

Reply:
Some of the functions use sequences. I'll change it to lists and do conversion at the call site